### DISCUSSION TITLE
Querying data where column has comma separated values

### DISCUSSION THREAD

**Discussion by TJMusiitwa** [supabase/supabase#12376] [Questions] [answered]
Hey Supabase community, 

### General Context
In my app, I want to query the speakers for the session but despite trying all filters and attempting to iterate over the values and parsing them into an int, I keep getting an error that mentions something to do with unknown big decimal.

The main idea is that when an end user is inputing information in a table or an external tool, they would just need to input the identifiers of the Speakers as comma separated and the app would be able to look up and display accordingly.

### Database Context
In one table called Speakers I have several fields including id of type **Int8** (Primary Key).
_This is generated by Supabase_

In another table Sessions, I have fields id, title and speakerIDs of type **Text** . An example of an input value in speakerIDs is **4,5**
_These values are entered based on the generated value from Supabase_

### Potential End Result
Is any one able to give me an idea of how I could structure the data such that the **speakerIDs** column is able to accept 1 or more values that I can then call from the Speaker table using filters. e.g. `.filter('id','in','(4,5)') `

### Sub- Question
Also can someone please explain to me what the Define as Array setting means in laymans terms. It mentions defines itself as:

> Allow column to be defined as variable-length multidimensional arrays

So does that mean that my column can be defined a list entry. 
There is no documentationt that explains or makes an example of how it is used or applied.

Thanks Supabase Community

**Comment by GaryAustin1** [comment, marked as answer]
You would normally use int[],bigint[], or text[] array columns and then use overlaps filter.  Overlaps compares an array column to an array of values and returns if any match.    In does a similar thing but on a single value column.    I know of no filter in the Supabase clients that will filter for one of several values in a comma separated sting.  

If you have array columns you can insert JavaScript arrays to them.  If that is what you are using.

**Reply by TJMusiitwa** [reply]
Hey @GaryAustin1 , so this question was coming from the perspective of a Flutter project. But after lots of debugging finally got the answer to my problem.
The Define as Array was just what I needed. And instead of using overlaps, I made use of the `in_` filter and done. 
Because Define as array returns my values as int8 values, it fit well with the filter.

I do however, feel it is lacking abit in documentation on its use and application.
Thanks again

**Reply by GaryAustin1** [reply]
If you have specific feedback on documentation for Supabase you can let them know better thru issues in this same github.
